docs/cloud/automation/terraform.md

```diff
@@ -1,2 +1,244 @@
+### Obtaining the `token`
+Currently this token must be created and displayed with the [esc cli][esc cli github releases] tool.  
+
+The token id displayed in the cloud console is not a valid token.
+
+The command to obtain a token from the esc cli tool is  
+`esc access tokens create --email <email>`
+
+## Resources
+
+All resources in Event Store Cloud can be provisioned using the Terraform provider.  
+Existing projects can be queried using a data source in the provider.  
+More complete samples can be found [here][terraform github samples].
+
+The following resources are available : 
+- Projects : `eventstorecloud_project`
+- Networks : `eventstorecloud_network`
+- Peering  : `eventstorecloud_peering`
+- Cluster  :  `eventstorecloud_managed_cluster`
+
+
+## eventstorecloud_project
+
+Looks up and creates project in the organization with which the provider is configured by name.
+### Arguments
+| Name       | Type   | Description                         |
+| :--------- | :----- | :-------------                      |
+| name       | string | *Required*, the name of the project |
+
+### Attributes
+| Name       | Type   | Description           | 
+| :--------- | :----- | :-------------        | 
+| id         | string | the ID of the project | 
+
+### Looking up a project
+
+<<< @/docs/cloud/automation/snippets/eventstorecloud_project.lookup.tf.hcl
+
+### Creating a project 
+
+<<< @/docs/cloud/automation/snippets/eventstorecloud_project.create.tf.hcl
+
+## eventstorecloud_network
+
+### Arguments
+| Name              | Type   | Description                                                                                      |
+| :---------        | :----- | :-------------                                                                                   |
+| name              | string | *Required*, the name of the network                                                              |
+| project_id        | string | *Required*, the ID of the project in which the network should be created                         |
+| resource_provider | string | *Required*, the name of the cloud (`aws`, `gcp`, `azure`) in which the network should be created |
+| region            | string | *Required*, the name of the region in which the network should be created                        |
+| cidr_block        | string | *Required*, the address space of the network.                                                    |
+
+### Attributes
+| Name       | Type   | Description           | 
+| :--------- | :----- | :-------------        | 
+| id         | string | the ID of the project | 
+
+::: tip
+`region` names must be in the format used by the resource provider, for example `us-west-2` for AWS, or `East US` for Azure
+:::
+
+::: tip
+`cidr_block`: the maximum prefix length is `/9` and  the minimum is `/24`.  
+However what is allowed is provider dependent.  
```

Comment:
**QUESTION:** Could you elaborate? Do you mean that the exact prefix length should be verified with the cloud provider constraints?